response.go

```diff
@@ -0,0 +1,248 @@
+	// When it is not present, the SignerIdentifier field in the SignerInfo
+	// will be used.
+	IssuerSerial issuerAndSerial `asn1:"optional"`
+}
+
+// eSSCertIDv2 uniquely identifies a certificate.
+//
+// Reference: RFC 5035 4
+//
+//	eSSCertIDv2 ::=  SEQUENCE {
+//	 hashAlgorithm           AlgorithmIdentifier
+//	 	DEFAULT {algorithm id-sha256},
+//	 certHash                 Hash,
+//	 issuerSerial             IssuerSerial OPTIONAL }
+type eSSCertIDv2 struct {
+	// HashAlgorithm is the hashing algorithm used to hash certificate.
+	// When it is not present, the default value is SHA256 (id-sha256).
+	// Supported values are SHA256, SHA384, and SHA512
+	HashAlgorithm pkix.AlgorithmIdentifier `asn1:"optional"`
+
+	// CertHash is the certificate hash using algorithm specified
+	// by HashAlgorithm. It is computed over the entire DER-encoded
+	// certificate (including the signature)
+	CertHash []byte
+
+	// IssuerSerial holds the issuer and serialNumber of the certificate.
+	// When it is not present, the SignerIdentifier field in the SignerInfo
+	// will be used.
+	IssuerSerial issuerAndSerial `asn1:"optional"`
+}
+
+// issuerAndSerial holds the issuer name and serialNumber of the certificate
+//
+// Reference: RFC 5035 4
+//
+//	IssuerSerial ::= SEQUENCE {
+//		issuer                   GeneralNames,
+//		serialNumber             CertificateSerialNumber }
+type issuerAndSerial struct {
+	IssuerName   generalNames
+	SerialNumber *big.Int
+}
+
+// generalNames holds the issuer name of the certificate.
+//
+// Reference: RFC 3280 4.2.1.7
+//
+// GeneralNames ::= SEQUENCE SIZE (1..MAX) OF GeneralName
+//
+//	GeneralName ::= CHOICE {
+//		otherName                       [0]     OtherName,
+//		rfc822Name                      [1]     IA5String,
+//		dNSName                         [2]     IA5String,
+//		x400Address                     [3]     ORAddress,
+//		directoryName                   [4]     Name,
+//		ediPartyName                    [5]     EDIPartyName,
+//		uniformResourceIdentifier       [6]     IA5String,
+//		iPAddress                       [7]     OCTET STRING,
+//		registeredID                    [8]     OBJECT IDENTIFIER }
+type generalNames struct {
+	Name asn1.RawValue `asn1:"optional,tag:4"`
+}
+
+// MarshalBinary encodes the response to binary form.
+// This method implements encoding.BinaryMarshaler.
+//
+// Reference: https://pkg.go.dev/encoding#BinaryMarshaler
+func (r *Response) MarshalBinary() ([]byte, error) {
+	if r == nil {
+		return nil, errors.New("nil response")
+	}
+	return asn1.Marshal(*r)
+}
+
+// UnmarshalBinary decodes the response from binary form.
+// This method implements encoding.BinaryUnmarshaler.
+//
+// Reference: https://pkg.go.dev/encoding#BinaryUnmarshaler
+func (r *Response) UnmarshalBinary(data []byte) error {
+	_, err := asn1.Unmarshal(data, r)
+	return err
+}
+
+// ValidateStatus validates the response.Status
+//
+// Reference: RFC 3161 2.4.2
+func (r *Response) ValidateStatus() error {
```

Comment:
In this file, some methods are declared as public even though they don't necessarily need to be. Once a method is made public, it cannot be reverted to private, but a private method can always be made public later if needed.

Reply:
Makes sense. Revisiting those methods and make unnecessary public methods to be private.